src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -528,6 +528,43 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 			}
 
 			var ok = OkPayload.Create(payload.Span, this);
+			if (m_rcbPolicyErrors != SslPolicyErrors.None)
+			{
+				// SSL would normally have thrown error, so connector need to ensure server certificates
+				// pass only if :
+				// * connection method is MitM-proof (e.g. unix socket)
+				// * auth plugin is MitM-proof and check SHA2(user's hashed password, scramble, certificate fingerprint)
+				if (cs.ConnectionProtocol != MySqlConnectionProtocol.UnixSocket)
+				{
+					if (string.IsNullOrEmpty(password) || !ValidateFingerprint(ok.StatusInfo, initialHandshake.AuthPluginData.AsSpan(0, 20), password!))
+					{
+						// fingerprint validation fail.
+						// now throwing SSL exception depending on m_rcbPolicyErrors
+						ShutdownSocket();
+						HostName = "";
+						lock (m_lock) m_state = State.Failed;
+						MySqlException ex;
+						switch (m_rcbPolicyErrors)
+						{
+							case SslPolicyErrors.RemoteCertificateNotAvailable:
+								// impossible
+								ex = new MySqlException(MySqlErrorCode.UnableToConnectToHost, "SSL not validated, no remote certificate available");
```

Comment:
This probably changes the `InnerException` that's being caught and rethrown? I want to compare this to the current behaviour.

Reply:
The simplest thing is probably to expose `m_rcbPolicyErrors` on an `AuthenticationException` used as the `.InnerException`.